### PR TITLE
egl: print supported dmabuf formats

### PR DESCRIPTION
**Pull request by agx**
Useful for debugging dmabuf related problems e.g. when a client requests a certain and we fail to support it.

### CODE REVIEW

File: render/egl.c [hunk 1]
```diff
@@ -101,6 +102,28 @@ static bool check_egl_ext(const char *egl_exts, const char *ext) {
 	return false;
 }
 
+static void print_dmabuf_formats(struct wlr_egl *egl)
+{
```
Comment (emersion) [inline render/egl.c]: Style nit: these two lines should be on the same one (I understand it's difficult to switch between wlroots' style and weston's :P ).

File: render/egl.c [hunk 2]
```diff
@@ -101,6 +102,28 @@ static bool check_egl_ext(const char *egl_exts, const char *ext) {
...
+		return;
+	}
+
+	int *formats;
+	int num = wlr_egl_get_dmabuf_formats(egl, &formats);
+	if (num < 0) {
+		return;
+	}
+
+	char *str_formats = malloc(num * 5 + 1);
```
Comment (emersion) [inline render/egl.c]: I'd prefer to use VLAs here:

```c
char str_formats[num * 5 + 1];
```

File: render/egl.c [hunk 3]
```diff
@@ -101,6 +102,28 @@ static bool check_egl_ext(const char *egl_exts, const char *ext) {
...
+	for (int i = 0; i < num; i++) {
+		sprintf(&str_formats[i*5], "%.4s ", (char*)&formats[i]);
```
Comment (emersion) [inline render/egl.c]: Use `snprintf` just in case.

**Comment by agx** [comment]
@emersion if that's the only style nit then I think I'm improving. Fixed.

**Comment by emersion** [comment]
Thanks!

**Pull request merged**